test/nft-collection.test.ts

```diff
@@ -1,88 +1,97 @@
-    // Check info of the minted NFT
-    expect(utils.checkHexString(nftMintedEventFields[2].value, nftName))
-    expect(utils.checkHexString(nftMintedEventFields[3].value, nftDescription))
-    expect(utils.checkHexString(nftMintedEventFields[4].value, nftUri))
-
-    expect(nftMintedEventFields[5].value).toEqual(nftContractId)
-    expect(nftMintedEventFields[6].value).toEqual(nftContractAddress)
-
-    const nftContractState = await fetchState(provider, NFT.contract, nftContractAddress, 0)
-
-    expect(nftContractState.fields.owner).toEqual(testAddress1)
-    expect(nftContractState.fields.isTokenWithdrawn).toEqual(true)
-    utils.checkHexString(nftContractState.fields.name, "CryptoPunk #0001")
-    utils.checkHexString(nftContractState.fields.description, "CP0001")
-    utils.checkHexString(nftContractState.fields.uri, "https://cryptopunks.app/cryptopunks/details/1")
-    expect(nftContractState.fields.collectionAddress).toEqual(nftCollectionContractAddress)
-
-    // Deposit NFT since it is withdrawn automatically when minted
-    await nftCollection.depositNFT(nftContractId)
-
-    // Burn NFT
-    const balanceBeforeBurning = await provider.addresses.getAddressesAddressBalance(testAddress1)
-
-    const gasAmount = 20000
-    const gasPrice = 1000000000 * 100
-    const totalGas = gasAmount * gasPrice
-    await nftCollection.burnNFT(
-      subContractId(nftCollectionContractId, stringToHex(nftUri), 0),
-      gasAmount,
-      BigInt(gasPrice)
-    )
-
-    const alphAmountInNFT = +nftContractState.asset.alphAmount.toString()
-    const balanceAfterBurning = await provider.addresses.getAddressesAddressBalance(testAddress1)
-    const refundedFromNFT = +balanceAfterBurning.balance + totalGas - +balanceBeforeBurning.balance
-
-    const relativeDiff = utils.relativeDiff(alphAmountInNFT, refundedFromNFT)
-    expect(relativeDiff).toBeLessThan(0.001)
-  }, 20000)
-})
+import { web3, subContractId, addressFromContractId, encodeU256, binToHex, groupOfAddress, addressFromTokenId } from '@alephium/web3'
+import * as utils from '../utils'
+import { NFTCollection } from '../utils/nft-collection'
+import { testWallet1 } from './signers'
+import { fetchNonEnumerableNFTState, fetchEnumerableNFTState } from '../utils/contracts'
+import { EnumerableNFTInstance, NFTOpenCollectionInstance, NFTPreDesignedCollectionInstance } from '../artifacts/ts'
+
+describe('nft collection', function() {
+  const nodeUrl = 'http://127.0.0.1:22973'
+  web3.setCurrentNodeProvider(nodeUrl)
+
+  it('should test minting nft in open collection', async () => {
+    const nftCollection = await getNFTCollection()
+    nftCollection.buildProject(false)
+
+    const nftCollectionDeployTx = await nftCollection.createOpenCollection("https://crypto-punk-uri")
+    const nftCollectionInstance = nftCollectionDeployTx.contractInstance
+
+
+    for (let i = 0n; i < 10n; i++) {
+      await mintOpenNFTAndVerify(nftCollection, nftCollectionInstance, i)
+    }
+  }, 60000)

```

Comment:
[Nit]: As the default `NodeProvider` has a rate limit, we can use`web3.setCurrentNodeProvider(nodeUrl, undefined, fetch)` to reduce testing time.

Reply:
Great tip, thanks!